simple_network_sim/loaders.py

```diff
@@ -127,6 +127,8 @@ def _check_overlap(one, two):
     Exception
     """
```

Comment:
I think we should simplify this function, assuming `x._lower < x._upper` (there's already an assertion for that), we could simplify this funtion to:

```
assert one._upper < two._lower or two._upper < one._lower, "overlapping ranges"
```

Reply:
There is no guarantee at the moment that `x._lower < x._upper`, only that `x._lower <= x._upper`. Since age ranges are defined with inclusive lower bounds and exclusive lower bounds, it seemed to make sense to allow `x._lower == x._upper`.

Also, those assertions also need to be `<=`, not `<`, as we expect upper and lower bounds to coincide (see the sample input file). However, if `one._upper == two._lower` and `one._lower == two._lower` we have overlapping age ranges, so we have to disallow age ranges of the form `[x,x)` to get a simple one-line function.

Should I disallow `[x,x)` type age ranges?

Reply:
I've adapted _check_overlap to the new form, and disabled `[x,x)`, in the latest commit now.

Reply:
True, `[x,x)` would be a problem, though I'm not sure that range makes sense from a mathematical perspective. We can't find it in the data we have, either, so I suggest we disallow `[x,x)` I suppose we also have to handle the case of one._upper being -1, which is infinite, essentially.